base/iterator.jl

```diff
@@ -348,6 +348,20 @@ length(t::Take) = _min_length(t.xs, 1:t.n, iteratorsize(t.xs), HasLength())
 
 start(it::Take) = (it.n, start(it.xs))
 
+"""
+    take(s::String, n)
+
+Return a `SubString` with at most `n` of the first `Char`s contained in `s`.
+"""
+function take(s::String, n::Int)
+    i = 1
+    while n > 1 && i < endof(s.data)
```

Comment:
`endof(s.data)` is equivalent to `length(s.data)`, but the latter is less confusing IMHO.